src/DataStructures/Variables.hpp

```diff
@@ -844,50 +844,67 @@ struct TempTensor {
 // @{
 /// \ingroup PeoGroup
 /// Variables Tags for temporary tensors inside a function.
-template <size_t N>
-using TempScalar = TempTensor<N, Scalar<DataVector>>;
+template <size_t N, typename DataType = DataVector>
+using TempScalar = TempTensor<N, Scalar<DataType>>;
```

Comment:
I'm confused about the change to these Tags. Perhaps I am missing something, but I don't see any uses of `TempX` buffers with non-`DataVector` data types... so why are the Tags being generalized?

Reply:
In future when we might need manipulate both real and complex DataVectors (or other non-primitive datatypes), these generalized Tags may come handy.

Reply:
That's fair enough. However, since this change to the Tags is unrelated to the change to add the shift derivs, it may be preferable to separate these changes into two commits. I say this because when I use `git log` to try to understand why the code in a file is structured in a particular way, I find the log _much_ easier to understand when logically unrelated changes are not made in these same commits...

Reply:
that makes sense. I've split changes into 2 commits.

Reply:
Thanks!